### PR TITLE
remove backticks and LDFLAG variable name to fix vulnerability

### PR DESCRIPTION
**Pull request by nikita-bhatia**
Following changes are for vulnerability fixes CVE-2023-24538 , CVE-2023-29404, CVE-2023-29405

1. Removed few backticks (“`” -chars) and replaced with `"` or `'`

2. Re the usage of LDFLAGS variable (we use it as a regular variable in a Makefile, no special meaning, since we’re not compiling any C/C++ code as a part of operator)

Note : Testing still needs to be done

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ script:
       export DOCKER_HUB_OPERATOR_TAG=23.7.0-dev
       export DOCKER_HUB_OPERATOR_TEST_TAG=23.7.0-dev
     else
-      export DOCKER_HUB_OPERATOR_TAG=`git rev-parse --short HEAD`
-      export DOCKER_HUB_OPERATOR_TEST_TAG=`git rev-parse --short HEAD`
+      export DOCKER_HUB_OPERATOR_TAG="git rev-parse --short HEAD"
+      export DOCKER_HUB_OPERATOR_TEST_TAG="git rev-parse --short HEAD"
```
Comment (zoxpx) [inline .travis.yml:18]: uh.. these are "shell-executable backticks" that actually "need to run"

So, we'd need to replace them with `$(...)` -->

```
      export DOCKER_HUB_OPERATOR_TAG=$(git rev-parse --short HEAD)
      export DOCKER_HUB_OPERATOR_TEST_TAG=$(git rev-parse --short HEAD)
```

File: pkg/util/k8s/k8s.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ const (
 )
 
 var (
-	kbVerRegex = regexp.MustCompile(`^(v\d+\.\d+\.\d+)(.*)`)
+	kbVerRegex = regexp.MustCompile("^(v\\d+\\.\\d+\\.\\d+)(.*)")
```
Comment (zoxpx) [inline pkg/util/k8s/k8s.go:54]: Good that you caught it.   👍

**Review by zoxpx — APPROVED** [review]
This looks OK -- thanks Nikita.

Please remember to update that `.travis.yml` -script

**Comment by zoxpx** [comment]
LOL -- this is too funny -- now the Travis and `staticcheck` are asking to return the "the executable-backticks" back:
```
go run github.com/golangci/golangci-lint/cmd/golangci-lint run --timeout=5m ./...
pkg/util/test/util.go:490:13: S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (gosimple)
					ver := regexp.MustCompile("\\S+\\/(\\d.\\S+)\\/version").FindStringSubmatch(env.Value)
					       ^
pkg/migration/generate.go:1189:20: S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (gosimple)
	validNameRegex := regexp.MustCompile("^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$")
	                  ^
pkg/util/util.go:100:25: S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (gosimple)
	hostnameWithDomainRe = regexp.MustCompile("^(?i)[a-z0-9-]+(\\.[a-z0-9-]+)+\\.?$")
	                       ^
```

We'll probably need to add `//lint:ignore S1007` to skip these 3 occurrences of the complaint.

File: .travis.yml (hunk 1 above)
Comment (nikita-bhatia) [reply .travis.yml]: updated

**Comment by zoxpx** [comment]
Looking good -- but, you might need to rebase the PR, to make the Travis happy...

**Comment by nikita-bhatia** [comment]
> Looking good -- but, you might need to rebase the PR, to make the Travis happy...

Travis is still failing because of same issue , after adding lint:ignore, staticcheck looks fine but lint is still complaining about same issue
```
 S1007: should use raw string (`...`) with regexp.MustCompile to avoid having to escape twice (gosimple)
					ver := regexp.MustCompile("\\S+\\/(\\d.\\S+)\\/version").FindStringSubmatch(env.Value) //lint:ignore S1007 we want to remove backticks to avoid vulnerability 
```

**Pull request closed without merging**